airbyte-webapp/src/core/form/schemaToYup.ts

```diff
@@ -0,0 +1,208 @@
+import { JSONSchema7 } from "json-schema";
+import * as yup from "yup";
+
+import { FormBlock, FormGroupItem, FormObjectArrayItem, FormConditionItem } from "core/form/types";
+import { isDefined } from "utils/common";
+
+import { FormBuildError } from "./FormBuildError";
+
+/**
+ * Returns yup.schema for validation
+ *
+ * This method builds yup schema based on jsonSchema ${@link JSONSchema7} and the derived ${@link FormBlock}.
+ * Every property is walked through recursively in case it is condition | object | array.
+ *
+ * @param jsonSchema
+ * @param formField The corresponding {@link FormBlock} to the given schema
+ * @param parentSchema used in recursive schema building as required fields can be described in parentSchema
+ * @param propertyKey used in recursive schema building for building path for uiConfig
+ * @param propertyPath constructs path of property
+ */
+export const buildYupFormForJsonSchema = (
+  jsonSchema: JSONSchema7,
+  formField: FormBlock,
+  parentSchema?: JSONSchema7,
+  propertyKey?: string,
+  propertyPath: string | undefined = propertyKey
+): yup.AnySchema => {
+  let schema:
+    | yup.NumberSchema
+    | yup.StringSchema
+    | yup.AnyObjectSchema
+    | yup.ArraySchema<yup.AnySchema>
+    | yup.BooleanSchema
+    | null = null;
+
+  if (jsonSchema.oneOf && propertyPath) {
+    const conditionFormField = formField as FormConditionItem;
+    // for all keys in all of the possible objects from the oneOf, collect the sub yup-schema in a map.
+    // the keys of the map are the keys of the property, the value is an array of the selection const value
+    // for the condition plus the sub schema for that property in that condition.
+    // As not all keys will show up in every condition, there can be a different number of possible sub schemas
+    // per key; at least one and at max the number of conditions (if a key is part of every oneOf)
+    const flattenedKeys: Map<string, Array<[unknown, yup.AnySchema]>> = new Map();
+    jsonSchema.oneOf.forEach((condition, index) => {
+      if (typeof condition !== "object") {
+        throw new FormBuildError("Spec uses oneOf with a condition that's not an object type");
+      }
+      const selectionConstValue = conditionFormField.selectionConstValues[index];
+      const selectionFormField = conditionFormField.conditions[index];
+      Object.entries(condition.properties || {}).forEach(([key, prop], propertyIndex) => {
+        if (!flattenedKeys.has(key)) {
+          flattenedKeys.set(key, []);
+        }
+        flattenedKeys
+          .get(key)
+          ?.push([
+            selectionConstValue,
+            typeof prop === "boolean"
+              ? yup.mixed()
```

Comment:
Why `yup.mixed()` instead of `yup.boolean()`?

Reply:
I inherited this from the old code. It didn't strike me as exceptionally important as we render a switch anyway for these so the user can't really do wrong on them in the first place.

Still, changed it to keep the code unsurprising.